apps/omg_api/lib/state/core.ex

```diff
@@ -221,24 +222,18 @@ defmodule OMG.API.State.Core do
     |> Map.new()
   end
 
-  # fee is implicit - it's the difference between funds owned and spend
-  defp amounts_add_up?(input_amounts, output_amounts, fees) do
-    outputs_covered =
-      for {output_currency, output_amount} <- Map.to_list(output_amounts) do
-        input_amount = Map.get(input_amounts, output_currency, 0)
-        input_amount >= output_amount
-      end
-      |> Enum.all?()
-
-    fees_covered =
-      for {input_currency, input_amount} <- Map.to_list(input_amounts) do
-        output_amount = Map.get(output_amounts, input_currency, 0)
-        fee = Map.get(fees, input_currency, 0)
-        input_amount - output_amount >= fee
-      end
-      |> Enum.any?()
-
-    if outputs_covered and fees_covered, do: :ok, else: {:error, :amounts_do_not_add_up}
+  defp amounts_add_up?(input_amounts, output_amounts) do
+    for {output_currency, output_amount} <- Map.to_list(output_amounts) do
+      input_amount = Map.get(input_amounts, output_currency, 0)
+      input_amount >= output_amount
+    end
+    |> Enum.all?()
+    |> if(do: :ok, else: {:error, :amounts_do_not_add_up})
+  end
+
+  def transaction_covers_fee?(input_amounts, output_amounts, fees) do
+    Transaction.Fee.covered?(input_amounts, output_amounts, fees)
+    |> if(do: :ok, else: {:error, :amounts_do_not_add_up})
```

Comment:
after this change maybe it's worth to separate these two error reports into two, explicitly? (i.e. `:fees_not_covered` in case of the latter). I remember both cases where merged into `:amounts_do_not_add_up` because they were indistinguishable for some time, but now they are distinguishable again